scopes/dependencies/dependency-resolver/dependency-resolver.docs.md

```diff
@@ -1,235 +1,322 @@
+
+- **Smart dependency installation**
+  The 'dependency-resolver' directs the [package manager](/docs/packages/overview) to install the right packages at the right place in the workspace file structure.
+  That includes installing multiple versions of the same package when different groups of components are set to use different versions.
+  The 'install' process includes importing components and linking them to the `node_modules` directory.
+
+## Quickstart & configuration
+
+### Selecting a package manager
+
+The Dependency resolver does not replace package managers - it uses them and directs them.
+To choose between '[Yarn](https://bit.dev/teambit/dependencies/yarn)' and '[pnpm](https://bit.dev/teambit/dependencies/pnpm)', set the `packageManager` property to either of the two:
+
+- `teambit.dependencies/yarn`
+- `teambit.dependencies/pnpm`
+
+```json
+// At the root-level of the workspace configuration JSON
+{
+  "teambit.dependencies/dependency-resolver": {
+    "packageManager": "teambit.dependencies/yarn"
+  }
+}
+```
+
+### Auto-registered dependency version and type
+
+Dependency policies define the version and dependency type of each package used by components in the workspace.
+Whenever you use the `bit install <package>` command, Bit registers the package version in the dependency configuration.
+
+```json
+// At the root-level of the workspace configuration JSON
+{
+  "@teambit.dependencies/dependency-resolver": {
+    "policy": {
+      "dependencies": {
+        "lodash": "4.17.0"
+      }
+    }
+  }
+}
+```
+
+### Applying policies on all relevant components
+
+A dependency policy set at the root level of the workspace configuration JSON will affect _all_ components that have the configured package as their dependency (i.e., components that have this module listed in their generated dependency graph).
+**Components that do not have this package as a dependency will not be affected.**
+
+For example:
+
+```json
+// At the root-level of the workspace configuration JSON
+
+// Every component that has 'lodash' as a dependency will use version '3.0.0' of it.
+// This policy will not affect any component that does not have 'lodash' as its dependency.
+
+{
+  "teambit.dependencies/dependency-resolver": {
+    "policy": {
+      "dependencies": {
+        "lodash": "3.0.0"
+      }
+    }
+  }
+}
+```
+
+### Applying policies on a selected group of components
+
+Dependency policies can be applied on a specific group of components. This is done using the [`@teambit.workspace/variants`](https://bit.dev/workspace/variants) configuration API.
+
+For example, to set the `1.0.0` version of `classnames` as a dependency of all components located inside the `./components/react` directory (or any of its sub-directories):
+
+```json
+{
+  "teambit.workspace/variants": {
+    "teambit.dependencies/dependency-resolver": {
+      "policy": {
+        "components/react": {
+          "classnames": "1.0.0"
```

Comment:
show an example of using - to remove something / or to remove and add it to another lifecycle